auacm/angular/src/app/models/problem.ts

```diff
@@ -0,0 +1,43 @@
+export class Problem {
+	constructor() {
```

Comment:
Maybe on all of these models, consider making a constructor that allows you to provide an `any` type.